vllm/distributed/parallel_state.py

```diff
@@ -633,6 +643,16 @@ def recv_tensor_dict(
                     # Skip broadcasting empty tensors.
                     _update_nested_dict(tensor_dict, key, tensor)
                     continue
+
+                # send-allgather: send only a slice, then do allgather.
+                tp_size = get_tensor_model_parallel_world_size()
+                if self.use_scatter_gather and tensor.numel() % tp_size == 0:
```

Comment:
We could introduce `maybe_use_scatter_gather = self.use_scatter_gather and tensor.numel() % tp_size == 0`, and use it in the following code as well.